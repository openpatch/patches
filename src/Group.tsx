import { Children, Fragment, ReactNode, useState } from "react";
import { css } from "@emotion/react";
import { Box } from "./Box";
import { Text } from "./Text";

export type GroupProps = {
  label: string;
  id?: string;
  variant?: "horizontal" | "vertical";
  children?: ReactNode;
};

export const Group = ({
  children,
  id,
  label,
  variant = "vertical",
}: GroupProps) => {
  const [genId] = useState(id || Date.now());
  const labelid = `group-${genId}`;
  return (
    <Fragment>
      <Text fontWeight="medium" id={labelid} mb="xsmall">
        {label}
      </Text>
      <Box
        role="group"
        aria-labelledby={labelid}
        css={[
          css`
            ul {
              list-style: none;
              margin-left: 0;
              padding-left: 0em;
              margin-bottom: 0;
              margin-top: 0;
            }
            li {
              margin: 8px 0;
            }

            li:last-of-type {
              margin-bottom: 0;
            }
          `,
          variant === "horizontal" &&
            css`
              ul {
                display: inline-flex;
                align-items: center;
                justify-content: center;
              }
              li {
                display: inline;
                margin: 0px 8px;
              }
            `,
        ]}
      >
        <ul>
          {Children.map(children, (c) => (
            <li>{c}</li>
          ))}
        </ul>
      </Box>
    </Fragment>
  );
};
