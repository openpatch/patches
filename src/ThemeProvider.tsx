import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
