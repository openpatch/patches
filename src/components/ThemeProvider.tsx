import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
