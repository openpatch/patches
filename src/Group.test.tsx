import { render } from "@testing-library/react";
import { Group, GroupProps } from "./Group";
import { ThemeProvider } from "./ThemeProvider";

Date.now = jest.fn(() => 1610568607249);
const renderWithTheme = (props?: GroupProps) =>
  render(
    <ThemeProvider>
      <Group {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
