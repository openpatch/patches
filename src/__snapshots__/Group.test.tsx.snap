// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  padding: 0px;
  margin: 0px;
  font-weight: 500;
  margin-bottom: 8px;
}

.emotion-1 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.emotion-1 ul {
  list-style: none;
  margin-left: 0;
  padding-left: 0em;
  margin-bottom: 0;
  margin-top: 0;
}

.emotion-1 li {
  margin: 8px 0;
}

.emotion-1 li:last-of-type {
  margin-bottom: 0;
}

<div>
  <div
    id="patches-root"
  >
    <p
      class="emotion-0"
      font-weight="medium"
      id="group-1610568607249"
    />
    <div
      aria-labelledby="group-1610568607249"
      class="emotion-1"
      role="group"
    >
      <ul />
    </div>
  </div>
</div>
`;
